src/back/glsl/mod.rs

```diff
@@ -769,6 +774,10 @@ impl<'a, W: Write> Writer<'a, W> {
         };
         writeln!(self.out, ";")?;
 
+        if add_a_newline {
```

Comment:
seems a bit weird to add this extra parameter and bit of logic, given that it can be done outside of this function. Is this to re-use it better?